### ISSUE TITLE
Aborted request still returning `loading: true`

### ISSUE THREAD

**Issue by ihmpavel** [apollographql/apollo-client#8575]
We use `apollo/client` in our CRA app with `react-router`. On the main page (url `/`) we want to load some data with `useQuery()` hook. The hook fires, but in the same moment our router redirect (changes) page to (url `/en`) with language (basename). Unfortunately, the hook has been fired already and cancelled. Because it is not full refresh, it will never be fired again. 

That means
1) Cancelled request in browser
2) `data` is `undefined`
3) `loading` is true

We have no way to fire the hook again. (`onError` or `onCompleted` are never called)

**Intended outcome:**

Property `loading` should be `false` after cancellation - abort controller.

**Actual outcome:**

Property `loading` is `true`.

**How to reproduce the issue:**

Create `CRA` app, create basename redirect with `react-router` and on main page create `useQuery` hook. Check `data`, `loading` variables.

**Versions**

```
  System:
    OS: Windows 10
  Binaries:
    Node: 14.16.0
  npmPackages:
    @apollo/client: I tried from v3.4.0 to the latest beta 3.5.0-beta.2
    apollo: ^2.33.4 => 2.33.4
    apollo-upload-client: ^16.0.0 => 16.0.0
```

**Comment by ihmpavel** [comment]
Update: Latest working version is `3.3.21`.

**Comment by brainkim** [comment]
@ihmpavel This might be fixed in 3.4.5. If not, a runnable reproduction would be helpful! Let me know.

**Comment by ihmpavel** [comment]
@brainkim It is working perfectly

Thank you ❤️
